### DISCUSSION TITLE
"Balanced" (MID) stream / RTMP protocol

### DISCUSSION THREAD

**Discussion by karlocikovic** [thirtythreeforty/neolink#258] [Ideas]
I have Reolink kit with Reolink NVR and 4x B800 cams, and 1x D800 cam. I decided to move on from Reolink and sell this whole kit, but then I found out Neolink and started testing.

Just short info about B800 (D800) streams available in Reolink NVR and Reolink app:

Main stream (Clear) = 3840x2160 (4K)
"Balanced" (Standard) = 896 x 516
Sub stream (Fluent) = 640 x 360

First I just streamed directly from Reolink NVR to BlueIris via RTSP protocol. I could just streamed  h.264 Substream (fluent). It is known that 4K cameras MainStream is using h.265 and is not possible to get good stream.

![image](https://user-images.githubusercontent.com/10189634/156556588-27ad68f0-0163-4362-ae55-34fdf209f8d1.png)

Then I tried to stream from Reolink NVR, but via RTMP protocol. There I could only see "Balanced" (Standard) stream. It is higher resolution than substream, and very fluid with no jerks. I am using this for few months in Blue Iris for Deepstack detection, and it is working perfectly.

![image](https://user-images.githubusercontent.com/10189634/156572695-95240c7a-a3ea-4277-af23-cd26ec67ef70.png)

Then I tried Neolink. First I tried with disconnecting B800 camera from Reolink NVR and connecting directly to my local network via POE switch. I configured Neolink as described on this site and managed to get good Main stream and Sub stream in Blue Iris. Receive Buffer is set to 20MB as suggested.

![image](https://user-images.githubusercontent.com/10189634/156574695-a99611a4-c51e-49da-a8db-d5657f939415.png)

I also tried to "pull" stream from Reolink NVR using Neolink.

![image](https://user-images.githubusercontent.com/10189634/156575323-e4163acc-04ba-45ed-ad61-dcbfc978dfc8.png)

But Neolink stream (Main or Sub) is not good as Balanced pulled out directly from NVR, there are too many jerks and drops. Do we have possibility to "pull out" Balanced stream from B800 cameras, as we can do with Main stream (Clear) and Sub stream (Fluent)?

**Comment by QuantumEntangledAndy** [comment]
This was made available in #196 as 'externStream'

**Reply by QuantumEntangledAndy** [reply]
There are some more details on how to use it in the issue thread that first requested it

https://github.com/thirtythreeforty/neolink/issues/189#issuecomment-899021744
